pkg/mcs/resource_manager/server/token_bukets.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2022 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,g
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	rmpb "github.com/pingcap/kvproto/pkg/resource_manager"
+)
+
+const defaultRefillRate = 10000
+
+const defaultInitialTokens = 10 * 10000
+
+// GroupTokenBucket is a token bucket for a resource group.
+type GroupTokenBucket struct {
+	*rmpb.TokenBucket `json:"token_bucket,omitempty"`
+	Consumption       *rmpb.TokenBucketsRequest `json:"consumption,omitempty"`
+	LastUpdate        *time.Time                `json:"last_update,omitempty"`
+	Initialized       bool                      `json:"initialized"`
+}
+
+// patch patches the token bucket settings.
+func (t *GroupTokenBucket) patch(settings *rmpb.TokenBucket) {
+	if settings == nil {
+		return
+	}
+	tb := proto.Clone(t.TokenBucket).(*rmpb.TokenBucket)
+	if settings.GetSettings() != nil {
```

Comment:
If we want to use burst, we should check whether brust is validate.

Reply:
Let us adding the validate logic in later, currently, I'm not sure the burst usage.

Reply:
@CabinfeverB  `userDefineHandlers` currently needs to be passed to etcd, and also we don't want let service to know the `pd.Server`, need to deconstruct it later.